### DISCUSSION TITLE
Adding new features for specific datasources

### DISCUSSION THREAD

**Discussion by agsha** [RedashCommunity/redash#14] [General]
Hi All,

Thanks for this fork and for breathing new life into Redash. Our company uses Redash heavily for all our BI needs and over the years we have many many fixes and new features to suit our organization. We feel many of these would be useful to the community and would like to get it merged.

Suppose we have added a new feature for a few specific data sources. For example - A small preview table icon, on click, shows about 100 rows from the table. For bigquery, this is a must-have feature, because the preview API is free of charge whereas "select * .. limit 100" will scan the entire table and cost $$ 

We have also enabled this feature for Postgres and MySQL because users liked it so much. 

But it is infeasible for us to add this support to each and every datasource supported by redash.

So my question is, can we send PRs that enable new features for only a few data sources and over time add support for other data sources as well?

**Comment by justinclift** [comment]
To me, it would make sense to have those features, as long as they're implemented in way which doesn't break or degrade the experience for others.

In theory (!), that's probably not too hard to do.  But pointing it out, just in case. :smile:
